### ISSUE TITLE
Error: Invalid tree found

### ISSUE THREAD

**Issue by mhilland** [broccolijs/broccoli#186]
Basically, I'm using the broccoli-render-templates plugin to compile my Dust.js templates, and need to pass it a data context. In order to achieve this, I wrote a function which loops through my models and creates an object containing all of my data. I had to write a promise function using bluebird in order to return all of this data back to my Brocfile.js so that the renderTemplates function could pass it to the compiler.

In order to ensure this data is available to the plugin, I need to place all of my chained events inside of the then() function of the promise. However, when I do this, I'm getting the following error:

<b>Error: Invalid tree found. You must supply a path or an object with a read function.</b>

Basically, any time I try to run my Brocfile.js chain in any other scope other than itself, I get this error. Why is this?

Here is all of my code working together for you to take a look at:

http://pastebin.com/FPYDL6Ea
